### PR TITLE
chore: add tp scheme adapter snippets. clarify auth wording

### PR DESCRIPTION
**Pull request by kleyow**
- add more snippets from the tp-scheme-adapter
- move away from "Auth" as a naming to avoid confusion between Authorization and Authentication (please double check that the are named correctly :sweat_smile: )

### CODE REVIEW

File: docs/thirdparty-openapi3-snippets.yaml
Comment (lewisdaly) [inline docs/thirdparty-openapi3-snippets.yaml:31]: This is part of the `Authorization` api resource, from the fspiop-api, so I would call this `AuthorizationChannelType` (ref: https://docs.mojaloop.io/mojaloop-specification/documents/API%20Definition%20v1.0.html#66-api-resource-authorizations)

File: docs/thirdparty-openapi3-snippets.yaml
Comment (lewisdaly) [reply docs/thirdparty-openapi3-snippets.yaml:45]: I'm also not sure if we want to use the term `channel`, since that term could get confusing, and isn't used elsewhere in the FSPIOP-API

File: docs/thirdparty-openapi3-snippets.yaml
Comment (lewisdaly) [inline docs/thirdparty-openapi3-snippets.yaml:292]: I don't see `currentState` on any of our sequence diagrams. (ref: https://raw.githubusercontent.com/mojaloop/pisp/master/docs/out/transfer/api_calls_detailed/PISPTransferDetailedAPI-page2.png)

File: docs/thirdparty-openapi3-snippets.yaml
Comment (lewisdaly) [inline docs/thirdparty-openapi3-snippets.yaml:66]: Maybe to avoid confusion we could call this `ConsentScopeType`

**Review by lewisdaly — CHANGES_REQUESTED** [review]
Thanks Kevin - this is important to clean up and make easier to understand. 

Here's some suggestions for naming conventions that might make things simpler:
- we should try to avoid the term `Authorization(s)`  in the thirdparty-api where possible - and keep `Authorization(s)` for the FSPIOP-API methods/objects etc. that we copied across
- For auth stuff related to `Consents` or `ConsentRequests`, maybe we can use Consent or Consent Request (e.g. `ConsentRequestChannelType` instead of `AuthorizationChannelType`

File: docs/thirdparty-openapi3-snippets.yaml
Comment (kleyow) [reply docs/thirdparty-openapi3-snippets.yaml:31]: done. thanks for clarifying

File: docs/thirdparty-openapi3-snippets.yaml
Comment (kleyow) [reply docs/thirdparty-openapi3-snippets.yaml:43]: done.

File: docs/thirdparty-openapi3-snippets.yaml
Comment (kleyow) [reply docs/thirdparty-openapi3-snippets.yaml]: We can address the usage of channel later.

File: docs/thirdparty-openapi3-snippets.yaml
Comment (kleyow) [reply docs/thirdparty-openapi3-snippets.yaml]: sounds good done.

File: docs/thirdparty-openapi3-snippets.yaml
Comment (kleyow) [reply docs/thirdparty-openapi3-snippets.yaml:291]: https://github.com/mojaloop/thirdparty-scheme-adapter/blob/master/src/interface/api-outbound-template.yaml#L306

copied from here. not sure where/why this was introduced

File: thirdparty/openapi3/schemas/AuthenticationType.yaml
Comment (kleyow) [reply thirdparty/openapi3/schemas/AuthenticationType.yaml]: done.

File: thirdparty/openapi3/schemas/AuthorizationStateType.yaml
Comment (kleyow) [reply thirdparty/openapi3/schemas/AuthorizationStateType.yaml:5]: https://github.com/mojaloop/thirdparty-scheme-adapter/blob/master/src/interface/api-outbound-template.yaml
Copied from the scheme-adapter. 
Seems to be isolated to the adapter? https://github.com/search?q=org%3Amojaloop+WAITING_FOR_AUTHORIZATION_REQUEST&type=code

**Comment by kleyow** [comment]
Thank you for clarifying and expanding on this Lewis. Glad we are ironing this out.

File: docs/thirdparty-openapi3-snippets.yaml
Comment (lewisdaly) [reply docs/thirdparty-openapi3-snippets.yaml:291]: Ok maybe @sridharvoruganti or @eoln will know more about this. My recommendation is that we remove it. It could be an internal state property that the thirdparty-scheme-adapter keeps track of, and perhaps exposes to the DFSP backend, but I don't think we need it in the thirdparty api.

File: docs/thirdparty-openapi3-snippets.yaml
Comment (lewisdaly) [reply docs/thirdparty-openapi3-snippets.yaml]: Sure, no biggie

File: docs/thirdparty-openapi3-snippets.yaml
Comment (eoln) [reply docs/thirdparty-openapi3-snippets.yaml:291]: @kleyow @lewisdaly currentState is only used in sync responses on Outbound

Comment (eoln) [reply docs/thirdparty-openapi3-snippets.yaml:291]: IMHO: The Outbound schemes elements should live in separate namespaces - maybe `scheme-adapters` or `SDK`  to distinguish them from the core switch interfaces

File: docs/thirdparty-openapi3-snippets.yaml
Comment (eoln) [inline docs/thirdparty-openapi3-snippets.yaml:55]: this is part of scheme-adapter

File: thirdparty/openapi3/schemas/AuthorizationStateType.yaml
Comment (eoln) [reply thirdparty/openapi3/schemas/AuthorizationStateType.yaml:5]: The Outbound responses are different from responses received from Switch. They differed because they contain details of state machine, also request will be different! they should be separated from the core Switch interfaces

File: thirdparty/openapi3/schemas/ThirdpartyAuthorizationIDPostResponse.yaml
Comment (eoln) [inline thirdparty/openapi3/schemas/ThirdpartyAuthorizationIDPostResponse.yaml:27]: currentState is only on Outbond. @sridharvoruganti could you point out Kevin to proper core interface?

**Review by eoln — CHANGES_REQUESTED** [review]
Scheme related interfaces are dedicated to synchronous operations (especially Outbound interfaces) which are merging gathering responses from asynchronous flows driven by state machines. They should be separated from core Switch interfaces. But the Scheme Inbound interfaces should match exactly the core Switch interfaces which should be implemented by DFSP/Thridparty

File: thirdparty/openapi3/schemas/AuthorizationStateType.yaml
Comment (kleyow) [reply thirdparty/openapi3/schemas/AuthorizationStateType.yaml:5]: Stupid questions incoming. Why is important to communicate the details of the state machine to another service?

Comment (kleyow) [reply thirdparty/openapi3/schemas/AuthorizationStateType.yaml:5]: If `currentState` is the only difference between outbound/switch interfaces, would using an optional `ExtensionList` be better for service specific stuff. Not sure but just tossing this out there.
@lewisdaly @eoln 
```
        extensionList:
          $ref: "#/components/schemas/ExtensionList"
          description: Optional extension, specific to deployment.
```

Lets try get this PR checked in before I work on the api doc generation tickets.

Comment (eoln) [reply thirdparty/openapi3/schemas/AuthorizationStateType.yaml:5]: other apis from sdk-scheme-adapters aren't using extension lists, so I followed this convention.

**Comment by kleyow** [comment]
Scheme related stuff removed.

File: docs/thirdparty-openapi3-snippets.yaml
Comment (eoln) [inline docs/thirdparty-openapi3-snippets.yaml]: we shouldn't use JSON.stringify as stringify method because it can generate non-identical strings, there is a need for additional design decision see: https://github.com/mojaloop/pisp/issues/66

**Review by eoln — APPROVED** [review]
lgtm, but let wait on @lewisdaly  before merge -> please take a look here also: https://github.com/mojaloop/pisp/issues/66

**Comment by kleyow** [comment]
The issue has been bubbled up to design. Going to merge this since it's blocking swagger documentation.

**Comment by HenkKodde** [comment]
hhhqqqqqw vbv gggggvhqqq1

**Pull request merged**
